ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs

```diff
@@ -55,9 +58,40 @@ private ReactChoreographer(CoreApplicationView applicationView)
             // main dispatcher or layout manager), yet it may not be yet accessible from the thread pool threads that'll soon
             // call ActivateCallback.
             _coreDispatcher = applicationView.Dispatcher;
+
+            if (!s_threadInstanceMap.TryAdd(_coreDispatcher, this))
+            {
+                throw new InvalidOperationException("Dispatcher thread already has a choreogrpaher");
+            }
+        }
+
+        /// <summary>
+        /// The choreographer instance corresponding to the calling dispatcher thread
+        /// </summary>
+        public static IReactChoreographer ThreadInstance
+        {
+            get
+            {
+                var currentView = CoreApplication.GetCurrentView();
+                if (currentView == null)
+                {
+                    throw new InvalidOperationException("ThreadInstance called from a non-dispatcher thread");
+                }
+
+                if (!s_threadInstanceMap.TryGetValue(currentView.Dispatcher, out var instance))
+                {
+                    throw new InvalidOperationException("Dispatcher thread has no choreographer");
+                }
+                return instance;
+            }
         }
 #else
         private ReactChoreographer() { }
+
+        /// <summary>
+        /// The choreographer instance (returns same as Instance for WPF)
+        /// </summary>
+       public static IReactChoreographer ThreadInstance => s_instance;
```

Comment:
Do we need `CreateSecondaryInstance` anymore?

Reply:
As opposed to calling ThreadInstance to construct it? IF we still keep Instance and ThreadInstance separate, would you still want ThreadInstance to create the instance (whereas Instance wouldn't), or not? I can go either way.

---
In reply to: [225741039](https://github.com/Microsoft/react-native-windows/pull/2000#discussion_r225741039) [](ancestors = 225741039)

Reply:
Correct - saying that CreateSecondaryInstance is more or less the same as ThreadInstance and can be removed, correct?

Reply:
Seems like ThreadInstance could just lazily create the instance. Any risk to that?

Reply:
Ok, will do lazy creation for the ThreadInstance.

Reply:
@rozele ah, I take that back, it's not that easy..
The existing usages of secondary choreographers are inside UIViewOperationQueueInstance where:
- constructor is called from LM thread 
- a bunch of choreographer method calls (ActivateCallback, etc.) are from other threads 
So I don't see any "good" place where I can hook a "_reactChoregrapher = ReactChoreographer.ThreadInstance"..